File: data/snippets/quiz014.cpp
class PaymentProcessor {
public:
    void processCreditCardPayment(double amount, const std::string& cardNumber, 
                               const std::string& expiryDate, const std::string& cvv) {
        // 验证信用卡详情
        if (cardNumber.length() != 16) {
            throw std::invalid_argument("Card number must be 16 digits");
        }
        
        if (!std::all_of(cardNumber.begin(), cardNumber.end(), ::isdigit)) {
            throw std::invalid_argument("Card number must contain only digits");
        }
        
        if (expiryDate.length() != 5 || expiryDate[2] != '/') {
            throw std::invalid_argument("Expiry date format must be MM/YY");
        }
        
        if (cvv.length() != 3 || !std::all_of(cvv.begin(), cvv.end(), ::isdigit)) {
            throw std::invalid_argument("CVV must be 3 digits");
        }
        
        // 处理付款
        std::cout << "Processing credit card payment of $" << amount << std::endl;
        std::cout << "Card details validated" << std::endl;
        std::cout << "Payment approved" << std::endl;
    }
    
    void processGiftCardPayment(double amount, const std::string& cardNumber, 
                             const std::string& expiryDate, const std::string& pin) {
        // 验证礼品卡详情 (几乎与信用卡验证相同)
        if (cardNumber.length() != 16) {
            throw std::invalid_argument("Card number must be 16 digits");
        }
        
        if (!std::all_of(cardNumber.begin(), cardNumber.end(), ::isdigit)) {
            throw std::invalid_argument("Card number must contain only digits");
        }
        
        if (expiryDate.length() != 5 || expiryDate[2] != '/') {
            throw std::invalid_argument("Expiry date format must be MM/YY");
        }
        
        if (pin.length() != 4 || !std::all_of(pin.begin(), pin.end(), ::isdigit)) {
            throw std::invalid_argument("PIN must be 4 digits");
        }
        
        // 处理付款
        std::cout << "Processing gift card payment of $" << amount << std::endl;
        std::cout << "Card details validated" << std::endl;
        std::cout << "Payment approved" << std::endl;
    }
    
    void processStoreCardPayment(double amount, const std::string& cardNumber, 
                              const std::string& expiryDate) {
        // 验证商店卡详情 (还是与信用卡验证相似)
        if (cardNumber.length() != 16) {
            throw std::invalid_argument("Card number must be 16 digits");
        }
        
        if (!std::all_of(cardNumber.begin(), cardNumber.end(), ::isdigit)) {
            throw std::invalid_argument("Card number must contain only digits");
        }
        
        if (expiryDate.length() != 5 || expiryDate[2] != '/') {
            throw std::invalid_argument("Expiry date format must be MM/YY");
        }
        
        // 处理付款
        std::cout << "Processing store card payment of $" << amount << std::endl;
        std::cout << "Card details validated" << std::endl;
        std::cout << "Payment approved" << std::endl;
    }
};

File: data/snippets/quiz014.java
public class PaymentProcessor {
    
    public void processCreditCardPayment(double amount, String cardNumber, 
                                       String expiryDate, String cvv) {
        // Validate credit card details
        if (cardNumber.length() != 16) {
            throw new IllegalArgumentException("Card number must be 16 digits");
        }
        
        if (!cardNumber.matches("\\d+")) {
            throw new IllegalArgumentException("Card number must contain only digits");
        }
        
        if (expiryDate.length() != 5 || expiryDate.charAt(2) != '/') {
            throw new IllegalArgumentException("Expiry date format must be MM/YY");
        }
        
        if (cvv.length() != 3 || !cvv.matches("\\d{3}")) {
            throw new IllegalArgumentException("CVV must be 3 digits");
        }
        
        // Process payment
        System.out.println("Processing credit card payment of $" + amount);
        System.out.println("Card details validated");
        System.out.println("Payment approved");
    }
    
    public void processDebitCardPayment(double amount, String cardNumber, 
                                      String expiryDate, String pin) {
        // Validate debit card details (almost identical to credit card validation)
        if (cardNumber.length() != 16) {
            throw new IllegalArgumentException("Card number must be 16 digits");
        }
        
        if (!cardNumber.matches("\\d+")) {
            throw new IllegalArgumentException("Card number must contain only digits");
        }
        
        if (expiryDate.length() != 5 || expiryDate.charAt(2) != '/') {
            throw new IllegalArgumentException("Expiry date format must be MM/YY");
        }
        
        if (pin.length() != 4 || !pin.matches("\\d{4}")) {
            throw new IllegalArgumentException("PIN must be 4 digits");
        }
        
        // Process payment
        System.out.println("Processing debit card payment of $" + amount);
        System.out.println("Card details validated");
        System.out.println("Payment approved");
    }
    
    public void processGiftCardPayment(double amount, String cardNumber, 
                                     String expiryDate, String pin) {
        // Validate gift card details (almost identical to credit card validation)
        if (cardNumber.length() != 16) {
            throw new IllegalArgumentException("Card number must be 16 digits");
        }
        
        if (!cardNumber.matches("\\d+")) {
            throw new IllegalArgumentException("Card number must contain only digits");
        }
        
        if (expiryDate.length() != 5 || expiryDate.charAt(2) != '/') {
            throw new IllegalArgumentException("Expiry date format must be MM/YY");
        }
        
        if (pin.length() != 4 || !pin.matches("\\d{4}")) {
            throw new IllegalArgumentException("PIN must be 4 digits");
        }
        
        // Process payment
        System.out.println("Processing gift card payment of $" + amount);
        System.out.println("Card details validated");
        System.out.println("Payment approved");
    }
    
    public void processStoreCardPayment(double amount, String cardNumber, 
                                      String expiryDate) {
        // Validate store card details (still similar to credit card validation)
        if (cardNumber.length() != 16) {
            throw new IllegalArgumentException("Card number must be 16 digits");
        }
        
        if (!cardNumber.matches("\\d+")) {
            throw new IllegalArgumentException("Card number must contain only digits");
        }
        
        if (expiryDate.length() != 5 || expiryDate.charAt(2) != '/') {
            throw new IllegalArgumentException("Expiry date format must be MM/YY");
        }
        
        // Process payment
        System.out.println("Processing store card payment of $" + amount);
        System.out.println("Card details validated");
        System.out.println("Payment approved");
    }
    
    public void processBusinessCardPayment(double amount, String cardNumber, 
                                         String expiryDate, String cvv, String businessId) {
        // Validate business card details (duplicate validation again)
        if (cardNumber.length() != 16) {
            throw new IllegalArgumentException("Card number must be 16 digits");
        }
        
        if (!cardNumber.matches("\\d+")) {
            throw new IllegalArgumentException("Card number must contain only digits");
        }
        
        if (expiryDate.length() != 5 || expiryDate.charAt(2) != '/') {
            throw new IllegalArgumentException("Expiry date format must be MM/YY");
        }
        
        if (cvv.length() != 3 || !cvv.matches("\\d{3}")) {
            throw new IllegalArgumentException("CVV must be 3 digits");
        }
        
        if (businessId == null || businessId.trim().isEmpty()) {
            throw new IllegalArgumentException("Business ID is required");
        }
        
        // Process payment
        System.out.println("Processing business card payment of $" + amount);
        System.out.println("Card details validated");
        System.out.println("Business ID verified");
        System.out.println("Payment approved");
    }
}
